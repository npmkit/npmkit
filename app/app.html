<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no" />
		<title>npmkit</title>
		<script>
			{
				if (!process.env.HOT) {
					const link = document.createElement('link');

					link.rel = 'stylesheet';
					link.href = '../build/bundle.css';

					document.write(link.outerHTML);
				}
			}
		</script>
	</head>
	<body>
		<main></main>
		<script>
			{
				if (process.env.NODE_ENV === 'development') {
				  const devToolsInstaller = require('electron-devtools-installer');

				  devToolsInstaller.default(devToolsInstaller.REACT_DEVELOPER_TOOLS).catch(() => {});
				}

				const script = document.createElement('script');

				script.src = (
					process.env.HOT ?
					'http://localhost:3000/build/bundle.js' :
					'../build/bundle.js'
				);

				document.write(script.outerHTML);
			}
		</script>
	</body>
</html>
