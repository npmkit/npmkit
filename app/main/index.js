import { setAppProtocol } from './MainProcessUtils';
import RootApplication from './RootApplication';

if (process.env.NODE_ENV === 'development') {
	require('electron-debug')(); // eslint-disable-line global-require
}

setAppProtocol();

// Keep only instance of root app
if (RootApplication.run().isShouldQuit()) {
	RootApplication.quit();
}
