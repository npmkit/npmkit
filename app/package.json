{
  "name": "npmkit",
  "productName": "npmkit",
  "version": "1.3.3",
  "description": "Tray app for quick access to all your projects",
  "license": "MIT",
  "repository": "sergeybekrin/npmkit",
  "homepage": "https://npmkit.app",
  "author": "Sergey Bekrin <sergey@bekrin.me> http://bekrin.me",
  "private": true,
  "main": "dist/index.main.js",
  "dependencies": {
    "electron-debug": "^2.0.0",
    "electron-is-dev": "^0.3.0",
    "electron-store": "^2.0.0",
    "electron-unhandled": "^1.1.0",
    "electron-util": "^0.9.0",
    "execa": "^0.10.0",
    "fix-path": "^2.1.0",
    "fuse.js": "^3.2.1",
    "lodash.debounce": "^4.0.8",
    "lodash.flatten": "^4.4.0",
    "menubar": "^5.2.3",
    "polished": "^1.9.3",
    "string-to-color": "^2.0.0",
    "styled-components": "^3.3.3",
    "thenby": "^1.2.6",
    "tree-kill": "^1.2.0",
    "unstated": "^2.1.1",
    "update-electron-app": "^1.3.0"
  }
}
