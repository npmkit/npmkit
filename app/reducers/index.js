import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';

import networkStatusReducer from './networkStatus';
import activeProjectReducer from './activeProjectCode';
import projectsFilterReducer from './projectsFilter';
import projectsReducer from './projects';
import scriptsReducer from './scripts';
import dependenciesReducer from './dependencies';
import npmReducer from './npm';
import nodeReducer from './node';
import moduleExplorerReducer from './moduleExplorer';
import localInstallerReducer from './localInstaller';

const rootReducer = combineReducers({

	// Common reducers
	routing: routerReducer,
	node: nodeReducer,
	npm: npmReducer,
	networkStatus: networkStatusReducer,
	projects: projectsReducer,
	projectsFilter: projectsFilterReducer,
	scriptsByIds: scriptsReducer,
	dependenciesByIds: dependenciesReducer,

	// Main app reducers
	activeProjectCode: activeProjectReducer,
	moduleExplorer: moduleExplorerReducer,

	// Protocol app reducers
	localInstaller: localInstallerReducer
});

export default rootReducer;
