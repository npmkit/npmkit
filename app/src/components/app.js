import { injectGlobal } from 'styled-components';
import TreyArrow from '~/components/trey-arrow';
import TreyPopup from '~/components/trey-popup';

injectGlobal`
  :root {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
    font-size: 16px;
    cursor: default;
  }

  body {
    padding: 0;
    margin: 0;
    margin-top: 0.3rem;
  }
`;

const App = () => (
  <React.Fragment>
    <TreyArrow />
    <TreyPopup>npmkit</TreyPopup>
  </React.Fragment>
);

export default App;
