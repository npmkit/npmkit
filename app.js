import { app, BrowserWindow } from 'electron';
import { name as protocol } from './package.json';

/**
 * @param {string} url
 * @return {boolean}
 */
// function isValidUrl (url) {
// 	let isUrl = false;
//
// 	try {
// 		new URL(url); // eslint-disable-line no-new
// 		isUrl = true;
// 	} catch (error) {
// 		isUrl = false;
// 	}
//
// 	return isUrl;
// }

/**
 * Global Application class
 */
export default class App {

	/**
	 * @type {?App}
	 */
	static instance = null;

	/**
	 * @type {?Object}
	 */
	window = null;

	/**
	 * @type {boolean}
	 */
	loaded = false;

	/**
	 * @type {Array<string>}
	 */
	urlQueue = [];

	/**
	 * @type {boolean}
	 */
	canQuite = false;

	/**
	 * Registers `npmkit://` protocol
	 */
	static setDefaultProtocol () {
		if (!app.isDefaultProtocolClient(protocol)) {
			app.setAsDefaultProtocolClient(protocol);
		}
	}

	/**
	 * Quite whole app
	 */
	static quit () {
		app.quit();
	}

	/**
	 * @return {App}
	 */
	static run () {
		if (App.instance === null) {
			App.instance = new App();
		}

		App.setDefaultProtocol();

		return App.instance;
	}

	/**
	 * Creates new app instance
	 */
	constructor () {

		// Open urls on hot start on macOS
		app.on('open-url', (event, url) => {
			event.preventDefault();

			this.openUrl(url);
		});

		// Quite app once all windows are closed
		app.on('window-all-closed', () => {
			if (process.platform !== 'darwin') {
				App.quit();
			}
		});

		// Show window on macOS once clicked in Dock
		app.on('activate', () => {
			this.window.show();
		});

		// Allow to quite app on macOS
		app.on('before-quite', () => {
			this.canQuite = true;
		});

		// Create window once app is ready
		app.on('ready', () => {
			const window = new BrowserWindow({
				show: false
			});

			window.loadURL(`file://${__dirname}/${process.env.NODE_ENV === 'development' ? 'app' : 'dist'}/app.html`);

			// Show and focus on app one loaded
			window.webContents.on('ready-to-show', () => {
				this.showAndFocus();
				this.loaded = true;

				// Open urls on cold start on macOS
				this.flushUrlQueue();
			});

			// Prevent window being closed on macOS
			window.on('close', (event) => {
				if ((process.platform !== 'darwin') ||
					(process.platform === 'darwin' && this.canQuite)) {
					this.window = null;
				} else {
					event.preventDefault();
					window.hide();
				}
			});

			// Open dev tools if required
			if (process.env.NODE_ENV === 'development') {
				window.openDevTools();
			}

			this.window = window;
		});
	}

	/**
	 * Opens dev tools
	 */
	openDevTools () {
		this.window.openDevTools();
	}

	/**
	 * Shows and sends focus to app
	 */
	showAndFocus () {
		this.window.show();
		this.window.focus();
	}

	/**
	 * @return {boolean}
	 */
	isShouldQuit () {
		return app.makeSingleInstance(commandLine => {

			// Open url on hot start on Windows
			this.openUrl(commandLine[1]);

			// Focus on window
			if (this.loaded) {
				if (this.window.isMinimized()) {
					this.window.restore();
				}

				this.showAndFocus();
			}
		});
	}

	/**
	 * Opens protocol url
	 *
	 * @todo(sbekrin) fix url check
	 */
	openUrl (url) {
		// if (!isValidUrl(url)) {
		// 	return;
		// }

		this.urlQueue.push(url);
		this.flushUrlQueue();
	}

	/**
	 * Opens urls in queue
	 */
	flushUrlQueue () {
		if (!this.loaded) {
			return;
		}

		while (this.urlQueue.length) {
			this.window.webContents.send('protocol-open', this.urlQueue.pop());
		}
	}
}
