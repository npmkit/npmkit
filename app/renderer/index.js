/* eslint-disable react/jsx-sort-props */
import 'babel-polyfill';

import React from 'react';
import { webFrame } from 'electron';
import { hashHistory, Router, Route, IndexRoute, IndexRedirect } from 'react-router';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { syncHistoryWithStore } from 'react-router-redux';

import { fixPath } from 'utils/ShellUtils';
import { setCurrentProject } from 'actions/ProjectActions';
import configureStore from 'store/store';
import Root from 'containers/Root';
import Application from 'components/Application';
import HomeView from 'containers/HomeView';
import SettingsView from 'containers/SettingsView';
import ProjectView from 'containers/ProjectView';
import ProjectInfo from 'containers/ProjectInfo';
import Scripts from 'containers/Scripts';
import Dependencies from 'containers/Dependencies';
import ProtocolApplication from 'containers/ProtocolApplication';
import LocalInstaller from 'containers/LocalInstaller';

import 'app.html';
import 'styles/root.styl';

const store = configureStore();
const history = syncHistoryWithStore(hashHistory, store);

// Prevent page zoom
webFrame.setZoomLevelLimits(1, 1);

// Fix PATH on OS X
fixPath();

// Handle project change
const handleProjectEnter = ({ params }) => store.dispatch(setCurrentProject(params.projectCode));
const handleProjectLeave = () => store.dispatch(setCurrentProject(null));

render(
	<Provider store={store}>
		<Router history={history}>
			<Route path='/' component={Root}>

				{/* Fix empty initial route */}
				<IndexRedirect to='main' />

				{/* Protocol app */}
				<Route path='protocol' component={ProtocolApplication}>
					<Route path='install' component={LocalInstaller} />
					<Route path='install-dev' component={LocalInstaller} />
					<Route path='install-optional' component={LocalInstaller} />
					{/* <Route path='install-global' component={GlobalInstaller} /> */}
				</Route>

				{/* Main app */}
				<Route path='main' component={Application}>

					{/* Home view */}
					<IndexRedirect to='/projects' />

					{/* Project view */}
					<Route path='/projects'>
						<IndexRoute component={HomeView} />
						<Route
							path=':projectCode'
							component={ProjectView}
							onEnter={handleProjectEnter}
							onLeave={handleProjectLeave}
						>
							<IndexRedirect to='info' />

							{/* Package project info */}
							<Route path='info' component={ProjectInfo} />

							{/* Project's npm scripts */}
							<Route path='tasks' component={Scripts} />

							{/* Project deps */}
							<Route path='dependencies' component={Dependencies} />
						</Route>
					</Route>

					{/* Global modules view */}
					<Route path='/modules' component={SettingsView} />

					{/* Settings view */}
					<Route path='/settings' component={SettingsView} />
				</Route>

			</Route>
		</Router>
	</Provider>,
	document.querySelector('main')
);
