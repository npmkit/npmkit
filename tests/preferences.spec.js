import { join } from 'path';
import { getRemoteApp } from './__fixtures__/app';

describe('Integration', () => {
  describe('preferences', () => {
    it('sets default preferences', async () => {
      const preferences = require(join(
        await getRemoteApp().getPath('userData'),
        'config.json'
      ));
      expect(preferences).toMatchSnapshot();
    });
  });
});
