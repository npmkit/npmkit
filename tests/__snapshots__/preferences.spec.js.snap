// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration preferences sets default preferences 1`] = `
Object {
  "editor": null,
  "pinned": Array [],
  "plugins": Array [],
  "projects": Array [],
  "terminal": "open -a \\"Terminal\\" %s",
}
`;
